File: python implementation/Code/naive.py
def search(pat, txt):
    M = len(pat)
    N = len(txt)

    # A loop to slide pat[] one by one */
    for i in range(N - M + 1):
        j = 0

        # For current index i, check
        # for pattern match */
        while(j < M):
            if (txt[i + j] != pat[j]):
                break
            j += 1

        if (j == M):
            print("Pattern found at index ", i)


if __name__ == '__main__':
    txt_file = input("Enter the path of the text file: ")
    pat_file = input("Enter the path of the pattern file: ")

    try:
        with open(txt_file, 'r') as f:
            txt = f.read().strip()
    except FileNotFoundError:
        print("Text file not found.")
        exit(1)

    try:
        with open(pat_file, 'r') as f:
            pat = f.read().strip()
    except FileNotFoundError:
        print("Pattern file not found.")
        exit(1)

    # Function call
    search(pat, txt)

File: C++ implementation/Code/naive.cpp
#include <iostream>
#include <fstream>
#include <string>
using namespace std;

void search(const string& pat, const string& txt) {
    int M = pat.size();
    int N = txt.size();

    /* A loop to slide pat[] one by one */
    for (int i = 0; i <= N - M; i++) {
        int j;

        /* For current index i, check for pattern match */
        for (j = 0; j < M; j++)
            if (txt[i + j] != pat[j])
                break;

        if (j == M) // if pat[0...M-1] = txt[i, i+1, ...i+M-1]
            cout << "Pattern found at index " << i << endl;
    }
}

int main() {
    string txt, pat;
    string txt_file, pat_file;

    cout << "Enter the path of the text file: ";
    cin >> txt_file;
    ifstream txt_input(txt_file);

    if (!txt_input.is_open()) {
        cout << "Unable to open file." << endl;
        return 1;
    }

    getline(txt_input, txt);

    txt_input.close();

    cout << "Enter the path of the pattern file: ";
    cin >> pat_file;
    ifstream pat_input(pat_file);

    if (!pat_input.is_open()) {
        cout << "Unable to open file." << endl;
        return 1;
    }

    getline(pat_input, pat);

    pat_input.close();

    // Function call
    search(pat, txt);
    return 0;
}
